File: M4 Intermediate DSA/src/L14_Sorting/Assignment1.java
package L14_Sorting;

import java.util.Arrays;

public class Assignment1 {

  public int solve(int[] A) {
    int N = A.length;
    int ans = 0;

    Arrays.sort(A);

    for (int index = N-1; index >= 0; index--) {
      ans += A[index] * (N - index);
    }

    return ans;
  }

}

File: M4_Intermediate_DSA/L3_Sorting/assignment1.go
package main

import "sort"

func SolveA1(A []int) int {
	ans := 0

	sort.Slice(A, func(i, j int) bool {
		return A[i] > A[j]
	})

	for i := range A {
		ans += A[i] * (i + 1)
	}

	return ans
}
